### PR TITLE
Add driver_package_override to launch_producer

### PR DESCRIPTION
**Pull request by rschiu**

### CODE REVIEW

File: cmd/launch_producer/main.cpp [hunk 1]
```diff
@@ -66,8 +67,29 @@ const char* kProducerPaths[] = {
 std::string getDriver() {
   std::string driver;
 
-  GET_STRING_PROP(kDriverProperty, driver);
-  LOG_INFO("Driver package: %s", driver.c_str());
+  // Read gapid.driver_package_override setting to override default property.
+  FILE* file = popen("settings get global gapid.driver_package_override", "r");
+  if (file != NULL) {
```
Comment (lpy) [inline cmd/launch_producer/main.cpp]: nullptr

File: cmd/launch_producer/main.cpp [hunk 2]
```diff
@@ -66,8 +67,29 @@ const char* kProducerPaths[] = {
...
+    const int BUF_LEN = 256;
+    char driver_override[BUF_LEN];
+    if (fgets(driver_override, BUF_LEN, file) != NULL) {
```
Comment (lpy) [inline cmd/launch_producer/main.cpp]: nullptr

File: cmd/launch_producer/main.cpp [hunk 3]
```diff
@@ -66,8 +67,29 @@ const char* kProducerPaths[] = {
 std::string getDriver() {
   std::string driver;
 
-  GET_STRING_PROP(kDriverProperty, driver);
-  LOG_INFO("Driver package: %s", driver.c_str());
+  // Read gapid.driver_package_override setting to override default property.
```
Comment (lpy) [inline cmd/launch_producer/main.cpp]: move this part to its own function `attemptToOverrideDriverPackage`.

Comment (rschiu) [reply cmd/launch_producer/main.cpp]: Done

File: cmd/launch_producer/main.cpp (hunk 2 above)
Comment (rschiu) [reply cmd/launch_producer/main.cpp]: Done

File: cmd/launch_producer/main.cpp (hunk 3 above)
Comment (rschiu) [reply cmd/launch_producer/main.cpp]: Done

File: cmd/launch_producer/main.cpp [hunk 4]
```diff
@@ -63,11 +64,35 @@ const char* kProducerPaths[] = {
...
-std::string getDriver() {
+std::string getDriverOverridePackage() {
   std::string driver;
+  // Read gapid.driver_package_override setting to override default property.
+  FILE* file = popen("settings get global gapid.driver_package_override", "r");
+  if (file != nullptr) {
+    const int BUF_LEN = 256;
+    char driver_override[BUF_LEN];
+    if (fgets(driver_override, BUF_LEN, file) != nullptr) {
+      LOG_INFO("Driver package override: %s", driver_override);
```
Comment (pmuetschard) [inline cmd/launch_producer/main.cpp]: Only log this if there actually is an override.

File: cmd/launch_producer/main.cpp [hunk 5]
```diff
@@ -63,11 +64,35 @@ const char* kProducerPaths[] = {
     "/lib/" ABI "/libgpudataproducer.so",
 };
 
-std::string getDriver() {
+std::string getDriverOverridePackage() {
```
Comment (pmuetschard) [inline cmd/launch_producer/main.cpp]: `getDriverPackageOverride` it's overriding the package.

File: cmd/launch_producer/main.cpp (hunk 4 above)
Comment (rschiu) [reply cmd/launch_producer/main.cpp]: Done

File: cmd/launch_producer/main.cpp (hunk 5 above)
Comment (rschiu) [reply cmd/launch_producer/main.cpp]: Done

File: cmd/launch_producer/main.cpp (hunk 1 above)
Comment (rschiu) [reply cmd/launch_producer/main.cpp]: Done

File: cmd/launch_producer/main.cpp [hunk 6]
```diff
@@ -63,11 +64,35 @@ const char* kProducerPaths[] = {
...
   std::string driver;
+  // Read gapid.driver_package_override setting to override default property.
+  FILE* file = popen("settings get global gapid.driver_package_override", "r");
+  if (file != nullptr) {
```
[the remark this answers is not in the document]
Comment (rschiu) [reply cmd/launch_producer/main.cpp]: Done

File: cmd/launch_producer/main.cpp [hunk 7]
```diff
@@ -63,11 +64,38 @@ const char* kProducerPaths[] = {
...
+std::string getDriverPackageOverride() {
   std::string driver;
+  // Read gapid.driver_package_override setting to override default property.
+  FILE* file = popen("settings get global gapid.driver_package_override", "r");
+  if (file == nullptr) {
+    return driver;
+  }
+  const int BUF_LEN = 256;
+  char driver_override[BUF_LEN];
+  if (fgets(driver_override, BUF_LEN, file) != nullptr) {
```
Comment (pmuetschard) [inline cmd/launch_producer/main.cpp]: Below we use:
```
char buffer[1024] = {0};
if (fgets(buffer, sizeof(buffer) - 1, fp) == nullptr)
```
I don't know which is better/more C++y, etc, and I don't really care either way. We can even keep both kinds... just thought I point it out.

Comment (rschiu) [reply cmd/launch_producer/main.cpp]: Didn't realize the other code.  I made some changes to make it more consistent.

**Pull request merged**